spark/wordcount/src/main/scala/com/example/bigtable/spark/wordcount/WordCount.scala

```diff
@@ -0,0 +1,154 @@
+    * https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/bigtable-client-1.0.0-pre2/bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java#L159
+    * @param config Hadoop config to set options on.
+    */
+  private def setBatchConfigOptions(config: Configuration) = {
+    config.set(BigtableOptionsFactory.BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL, "true")
+
+    // Dataflow should use a different endpoint for data operations than online traffic.
+    config.set(BigtableOptionsFactory.BIGTABLE_HOST_KEY, BigtableOptions.BIGTABLE_BATCH_DATA_HOST_DEFAULT)
+
+    config.set(BigtableOptionsFactory.INITIAL_ELAPSED_BACKOFF_MILLIS_KEY, String.valueOf(TimeUnit.SECONDS.toMillis(5)))
+
+    config.set(BigtableOptionsFactory.MAX_ELAPSED_BACKOFF_MILLIS_KEY, String.valueOf(TimeUnit.MINUTES.toMillis(5)))
+
+    // This setting can potentially decrease performance for large scale writes. However, this
+    // setting prevents problems that occur when streaming Sources, such as PubSub, are used.
+    // To override this behavior, call:
+    //    Builder.withConfiguration(BigtableOptionsFactory.BIGTABLE_ASYNC_MUTATOR_COUNT_KEY,
+    //                              BigtableOptions.BIGTABLE_ASYNC_MUTATOR_COUNT_DEFAULT);
+    config.set(BigtableOptionsFactory.BIGTABLE_ASYNC_MUTATOR_COUNT_KEY, "0")
+  }
+
+
+  /**
+    * Create a table in the Cloud Bigtable instance if it doesn't already
+    * exit.
+    *
+    * @param connection A Cloud Bigtable connection
+    * @param name       The table name
+    */
+  def createTableIfNotExists(connection: Connection, name: String) = {
+    val tableName = TableName.valueOf(name)
+    val admin = connection.getAdmin()
+    try {
+      if (!admin.tableExists(tableName)) {
+        val tableDescriptor = new HTableDescriptor(tableName)
+        tableDescriptor.addFamily(
+          new HColumnDescriptor(ColumnFamily))
+        admin.createTable(tableDescriptor)
+      }
+    } finally {
+      admin.close()
+    }
+  }
+
+  /**
+    * Main entry point for running the WordCount spark job and writing
+    * the results to Bigtable.
+    *
+    * @param projectId  The Google Cloud Project ID
+    * @param instanceId The Cloud Bigtable instance name
+    * @param tableName  The name of the Cloud Bigtable table to write to
+    * @param fileName   The file (local or gcs) to sort
+    * @param sc         The Spark context
+    */
+  def runner(projectId: String,
+             instanceId: String,
+             tableName: String,
+             fileName: String,
+             sc: SparkContext) = {
+    val createTableConnection = createConnection(projectId, instanceId)
+    try {
+      createTableIfNotExists(createTableConnection, tableName)
+    } finally {
+      createTableConnection.close()
+    }
+
+    var conf = BigtableConfiguration.configure(
+      projectId, instanceId)
+    conf.set(TableInputFormat.INPUT_TABLE, tableName)
```

Comment:
Can you please remove out this line:
` conf.set(TableInputFormat.INPUT_TABLE, tableName)`
`TableOutputFormat.OUTPUT_TABLE` is used, and not the `INPUT_TABLE` setting, since you're using a `TableOutputFormat` (in line 127).